include/exec/system_scheduler.hpp

```diff
@@ -0,0 +1,663 @@
+    return rhs_impl && rhs_impl->ctx_ == ctx_;
+  }
+};
+
+struct __exec_system_operation_state_impl;
+using __exec_pool_sender_t = decltype(stdexec::schedule(std::declval<__exec_system_scheduler_impl>().pool_scheduler_));
+
+struct __exec_system_pool_receiver {
+  using receiver_concept = stdexec::receiver_t;
+
+  friend void tag_invoke(stdexec::set_value_t, __exec_system_pool_receiver&&) noexcept;
+
+  friend void tag_invoke(stdexec::set_stopped_t, __exec_system_pool_receiver&&) noexcept;
+
+  friend void tag_invoke(stdexec::set_error_t, __exec_system_pool_receiver&&, std::exception_ptr) noexcept;
+
+  friend stdexec::empty_env tag_invoke(stdexec::get_env_t, const __exec_system_pool_receiver&) noexcept {
+    return {};
+  }
+
+  __exec_system_operation_state_impl* os_ = nullptr;
+};
+
+struct __exec_system_operation_state_impl : public __exec_system_operation_state_interface {
+  __exec_system_operation_state_impl(
+    __exec_pool_sender_t&& pool_sender,
+    __exec_system_receiver&& recv) :
+    recv_{std::move(recv)},
+    pool_operation_state_{
+      [&](){return stdexec::connect(std::move(pool_sender), __exec_system_pool_receiver{this});}()} {
+  }
+
+  __exec_system_operation_state_impl(const __exec_system_operation_state_impl&) = delete;
+  __exec_system_operation_state_impl(__exec_system_operation_state_impl&&) = delete;
+  __exec_system_operation_state_impl& operator= (const __exec_system_operation_state_impl&) = delete;
+  __exec_system_operation_state_impl& operator= (__exec_system_operation_state_impl&&) = delete;
+
+
+  void start() noexcept override {
+    stdexec::start(pool_operation_state_);
+  }
+
+  __exec_system_receiver recv_;
+  decltype(stdexec::connect(
+      std::move(std::declval<__exec_pool_sender_t>()), std::move(std::declval<__exec_system_pool_receiver>())))
+    pool_operation_state_;
+};
+
+inline void tag_invoke(stdexec::set_value_t, __exec_system_pool_receiver&& recv) noexcept {
+  __exec_system_receiver &system_recv = recv.os_->recv_;
+  system_recv.set_value((system_recv.cpp_recv_));
+}
+
+inline void tag_invoke(stdexec::set_stopped_t, __exec_system_pool_receiver&& recv) noexcept {
+  __exec_system_receiver &system_recv = recv.os_->recv_;
+  recv.os_->recv_.set_stopped(&(system_recv.cpp_recv_));
+}
+
+inline void tag_invoke(stdexec::set_error_t, __exec_system_pool_receiver&& recv, std::exception_ptr ptr) noexcept {
+  __exec_system_receiver &system_recv = recv.os_->recv_;
+  recv.os_->recv_.set_error(&(system_recv.cpp_recv_), &ptr);
+}
+
+
+struct __exec_system_sender_impl : public __exec_system_sender_interface {
+  __exec_system_sender_impl(
+        __exec_system_scheduler_impl* scheduler, __exec_pool_sender_t&& pool_sender) :
+      scheduler_{scheduler}, pool_sender_(std::move(pool_sender)) {
+
+  }
+
+  __exec_system_operation_state_interface* connect(__exec_system_receiver recv) noexcept override {
+    return
+      new __exec_system_operation_state_impl(std::move(pool_sender_), std::move(recv));
```

Comment:
there are a slew of `new`'s and no `delete`s or `unique_ptr`s. are all the allocations intended to leak?

Reply:
I would mark all these as things still left to be done. The implementation is somehow between Phase 1 and Phase 2 (using Lee's plan).

Reply:
I'm fine leaving phases 2 and 3 to later. i'm not fine with known memory leaks in `main`.